### PR TITLE
[Task04]

### PR DESCRIPTION
**Pull request by ando-andy**
## Related Tickets

- [#Ticket ID](https://docs.google.com/spreadsheets/d/1F5V2sIUMGej3Wrp8qYRscf0KaVSlccBmqkRx49_hUVo/edit#gid=935273790)

## What's this PR do ?
- Category screen created

## Library
*(List gem, library third party add new)*
https://github.com/WenchaoD/FSPagerView
https://github.com/tuan188/MGCleanArchitecture

## Impacted Areas in Application
*(List features, api, models or services that this PR will affect)*

## Checklist

- [x] It was tested in local success ?
- [x] Updated rake task, environment variable at [here](https://???)
- [x] Fill link PR into ticket and the opposite
- [x] Note reason, scope of influence, solution into ticket
- [x] Validate UI/Model/API

## Notes
*(Other notes)*

### CODE REVIEW

File: MovieDB/Scenes/Categories/CategoriesViewModel.swift [hunk 1]
```diff
@@ -6,22 +6,71 @@
...
     func transform(_ input: Input) -> Output {
-        return Output()
+        let loadTrigger = input.loadTrigger
+            .map { _ in self.category }
+        let reloadTrigger = input.reloadTrigger
+            .withLatestFrom(loadTrigger)
+        let loadMoreTrigger = input.loadMoreTrigger
+            .withLatestFrom(loadTrigger)
+        
+        let loadMoreOutput = setupLoadMorePagingWithParam(loadTrigger: loadTrigger,
```
Comment (longvt-1468) [inline MovieDB/Scenes/Categories/CategoriesViewModel.swift]: let configOutput = configPagination(
            loadTrigger: input.loadTrigger,
            reloadTrigger: input.reloadTrigger,
            getItems: {
                self.useCase.getMoviesList(category: self.categry)
            },
            loadMoreTrigger: input.loadMoreTrigger,
            loadMoreItems: { page in
                self.useCase.loadMoreMoviesList(category: self.category, page: page)
            })

File: MovieDB/Scenes/Categories/Cell/CategoriesTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import UIKit
+
+final class CategoriesTableViewCell: UITableViewCell, NibReusable {
+
+// MARK: - IBOutlets
+    @IBOutlet weak var posterImage: UIImageView!
+    @IBOutlet weak var nameLabel: UILabel!
+    
+    func bindViewModel(_ viewModel: MovieViewModel?) {
+        nameLabel.text = viewModel?.name ?? ""
```
Comment (longvt-1468) [inline MovieDB/Scenes/Categories/Cell/CategoriesTableViewCell.swift]:
```suggestion
        nameLabel.text = viewModel?.name
```

Comment (ando-andy) [reply MovieDB/Scenes/Categories/Cell/CategoriesTableViewCell.swift]: Modified🙇‍♂️

File: MovieDB/Scenes/Categories/CategoriesViewModel.swift (hunk 1 above)
Comment (ando-andy) [reply MovieDB/Scenes/Categories/CategoriesViewModel.swift]: Changed & Removed

**Pull request merged**